### PR TITLE
Rename doc files and fix typos

### LINKED ISSUE #238: Use `lowercase_with_underscore.rst`  naming convention for doc files
Creating an issue for now:

---
@sbillinge Do we want to use `lower_with_underscore` for doc .rst files? A bit hard to read `diffractionobjectsexample.rst`. Not sure about the convention. Did a quick search and appears `NumPy` is using the `lower_with_underscore` convention (mostly, not all) https://github.com/numpy/numpy/tree/main/doc/source/reference

@bobleesj @yucongalicechen yes, let's use this convention.  But I don't worry too much for now, I will just merge this.

_Originally posted by @sbillinge in https://github.com/diffpy/diffpy.utils/issues/237#issuecomment-2544131453_

### PR DESCRIPTION
**Pull request by yucongalicechen**
closes #238
@bobleesj I can have some feedback before @sbillinge reviews it

### CODE REVIEW

File: doc/source/examples/transforms_example.rst [hunk 1]
```diff
@@ -13,27 +13,27 @@ This example will demonstrate how to use the functions in the
    to convert it to ``2theta`` or ``d``. ::
 
     # Example: convert q to 2theta
-    from diffpy.utils.transformers import q_to_tth
+    from diffpy.utils.transforms import q_to_tth
```
Comment (yucongalicechen) [inline doc/source/examples/transforms_example.rst:16]: fix typo

**Comment by bobleesj** [comment]
> closes #238
> 
> @bobleesj I can have some feedback before @sbillinge reviews it

Have you rendered the docs locally to check all links are populated and no broken links?

**Comment by yucongalicechen** [comment]
> > closes #238
> > @bobleesj I can have some feedback before @sbillinge reviews it
> 
> Have you rendered the docs locally to check all links are populated and no broken links?

Yep, I just built them locally and every link works. Here's a screenshot:

**Comment by yucongalicechen** [comment]
The transforms example was displayed improperly. I edited it and it's working now.

**Comment by yucongalicechen** [comment]
@bobleesj in the docstring the XQUANTITIES is not showing. Shall I open a new issue or just comment on #239?

**Comment by bobleesj** [comment]
> @bobleesj in the docstring the XQUANTITIES is not showing. Shall I open a new issue or just comment on #239?
> 
yes please open the issue!

**Comment by bobleesj** [comment]
Lovely, looks good to me

File: doc/source/examples/transforms_example.rst [hunk 2]
```diff
@@ -10,44 +10,50 @@ This example will demonstrate how to use the functions in the
...
 
 (3) Converting from ``d`` to ``q`` or ``2theta``:
-    For a 1D ``d`` array, you can convert it to ``q`` or ``2theta``. ::
+    For a 1D ``d`` array, you can convert it to ``q`` or ``2theta``.
+
+.. code-block:: python
 
     # Example: convert d to q
-    from diffpy.utils.transformers import tth_to_q
+    from diffpy.utils.transforms import d_to_q
```
Comment (bobleesj) [inline doc/source/examples/transforms_example.rst:51]: good catch

File: doc/source/utilities/utilities.rst [hunk 3]
```diff
@@ -10,6 +10,7 @@ Check the :ref:`examples<Examples>` provided for how to use these.
 .. contents::
    :depth: 2
 
-.. include:: parsersutility.rst
-.. include:: resampleutility.rst
-.. include:: toolsutility.rst
+.. include:: parsers_utility.rst
+.. include:: resample_utility.rst
+.. include:: tools_utility.rst
+.. include:: transforms_utility.rst
```
Comment (bobleesj) [inline doc/source/utilities/utilities.rst:16]: nice

**Comment by yucongalicechen** [comment]
thanks @bobleesj ! @sbillinge ready to review now!

**Comment by sbillinge** [comment]
> > @bobleesj in the docstring the XQUANTITIES is not showing. Shall I open a new issue or just comment on #239?
> 
> yes please open the issue!

This may not fix.  We will have to see.  It depends if that unpack can be made to be done dynamically during the doc builds or not.  If it doesn't fix it is honestly OK to have it like this in the API docs which are generally "developer-facing" and  not "user-facing" and I would expect a developer to be able to figure out what is going on

**Pull request merged**